app/models/user.rb

```diff
@@ -0,0 +1,28 @@
+class User < ApplicationRecord
+  has_secure_password
+
+  before_save :downcase_email
+
+  validates :email, format: { with: URI::MailTo::EMAIL_REGEXP }, presence: true, uniqueness: true
+  validates :password, length: { within: 8..40 }
```

Comment:
More of a nitpick that has nothing to do with code: Allow long passwords. In SQL there is no storage different (if I recall) between 40 chars and 255. 255 is the minimum length of varchar column. Just let me have super long password manager managed passwords 😄